### DISCUSSION TITLE
Serenity.Criteria.and Question

### DISCUSSION THREAD

**Discussion by kilroyFR** [serenity-is/Serenity#6255] [General]
hey there, 

i have a query performed on TS side.

My filters are in 3 parts. First 2 are filter on a value and on 'is null' condition
But for the last filter i would like to create a formula that looks like something like this when requested to Oracle DB
`"SCHEDULED_DATE + (5/1440) < SYSDATE"`

```
OK>        request.Criteria = Serenity.Criteria.and(['is null', [fld.ExecutedDate]], request.Criteria);
OK>        request.Criteria = Serenity.Criteria.and([[fld.Status], '=', 0], request.Criteria);
KO<        request.Criteria = Serenity.Criteria.and([[fld.ScheduledDate] + ' + (5/1440)', '<' , 'SYSDATE'], request.Criteria);
```

Generated query looks like this one 

`"WHERE (((:p1 < :p2) AND ((T0.STATUS = :p3) AND T0.EXECUTED_DATE IS NULL)))"`

while i would have preferred

`"WHERE (((:p1 + (5/1440) < :p2) AND ((T0.STATUS = :p3) AND T0.EXECUTED_DATE IS NULL)))"`

Anyone else did encounter that kind of need ?

Looks like it's impossible as it requires a Field name first then a valid operation then a value. 
My first parameter is a calculation on a column, not a simple column itself.

Thanks for any help

**Comment by kilroyFR** [comment]
Well i simplified the calculation moving the calculation to the filter value.

```
var dt = new Date();
dt.setMinutes(dt.getMinutes() - 5);
 request.Criteria = Serenity.Criteria.and([[fld.ScheduledDate], '<' , dt], request.Criteria);
```

**Reply by VictorTomaili** [reply]
Formules are risky to send from the script side to the server side. Good to know solved by adding final value to the criteria.
